gulp/php.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-env es6 */
+'use strict';
+
+// External dependencies
+import requireUncached from 'require-uncached';
+import pump from 'pump';
+import {src, dest} from 'gulp';
+import log from 'fancy-log';
+import colors from 'ansi-colors';
+
+// Internal dependencies
+import {paths, rootPath, gulpPlugins, gulpReplaceOptions} from './constants';
+
+// We are on the first run by default
+let isFirstRun = true;
+
+// Grab the initial config so we can check if the theme slug/name change later
+let initialConfig = require(`${rootPath}/dev/config/themeConfig.js`);
+
+// Stash theme config
+let themeConfig = initialConfig.theme;
+
+/**
+ * PHP via PHP Code Sniffer.
+ */
+export default function php(done) {
+
+    // get a fresh copy of the config
+    const config = requireUncached(`${rootPath}/dev/config/themeConfig.js`);
+
+	// We should rebuild if this is the first run OR the theme slug/name have changed
+	let isRebuild = isFirstRun ||
+		( themeConfig.slug !== config.theme.slug ) ||
+        ( themeConfig.name !== config.theme.name );
+
+	if ( isRebuild ) {
+		themeConfig.slug = config.theme.slug;
+        themeConfig.name = config.theme.name;
+        log(colors.yellow(`Rebuilding ${colors.bold('all')} the PHP files, this may take a while...`));
```

Comment:
Excellent job @ataylorme adding the log messages here.  Very nice 👍

Reply:
Thanks for the +1 @hellofromtonya. It actually isn't too slow for me with the base PHP files but I can imagine it slowing down as more files are added during theme development.

Reply:
It isn't slow for me either.  But the indication in the console helps tell a story of what's going on.  I like that.